crates/oxc_transformer/src/es2015/shorthand_properties.rs

```diff
@@ -0,0 +1,53 @@
+use oxc_ast::{ast::*, AstBuilder};
+use oxc_span::GetSpan;
+
+use std::rc::Rc;
+
+/// ES2015: Shorthand Properties
+///
+/// References:
+/// * <https://babel.dev/docs/babel-plugin-transform-shorthand-properties>
+/// * <https://github.com/babel/babel/blob/main/packages/babel-plugin-transform-shorthand-properties>
+pub struct ShorthandProperties<'a> {
+    ast: Rc<AstBuilder<'a>>,
+}
+
+impl<'a> ShorthandProperties<'a> {
+    pub fn new(ast: Rc<AstBuilder<'a>>) -> Self {
+        Self { ast }
+    }
+
+    pub fn transform_object_property<'b>(&mut self, obj_prop: &'b mut ObjectProperty<'a>) {
+        if !obj_prop.shorthand && !obj_prop.method {
+            return;
+        }
+
+        obj_prop.shorthand = false;
+        obj_prop.method = false;
+
+        if obj_prop.computed {
+            return;
+        }
+
+        let is_proto_id = obj_prop.key.is_specific_id("__proto__");
+        let is_proto_string = obj_prop.key.is_specific_string_literal("__proto__");
+
+        if !is_proto_id && !is_proto_string {
+            return;
+        }
+
+        obj_prop.computed = true;
```

Comment:
Same here, mutating the value then return early?

Reply:
I found a bug here. Let me fix it.